### ISSUE TITLE
`no-extra-parens`: request an option to ignore the multiline variables

### ISSUE THREAD

**Issue by Yong-Zen-Xiang** [eslint-stylistic/eslint-stylistic#872]
### Clear and concise description of the problem

Sometimes when the variables is exceed the max length or we want to improve readability, we need to break down the variables, for example

**Conditional Statements**
```
const conditionStatement = 
  condition 1 &&
  condition 2 &&
  condition 3;
``` 

**Chaining Statements**
```
const joinedText = 
  dataFromQuery
    .filter((item) => item.isActive)
    .map((item) => item.name)
    .join("");
```

But when we enabled the `operator-linebreak`, it will pop-up the error, `There should be no line break before or after '='`

So to solve this, I would like to use the parenthesis to keeping the readability, for example

```
const conditionStatement = (
  condition 1 &&
  condition 2 &&
  condition 3
);
``` 

But `no-extra-parens` will pop-up error for the code above, and the combination with `operator-linebreak` will auto fix the code to below

```
const conditionStatement = condition 1 &&
  condition 2 &&
  condition 3;
```

Will break the readability if like this 😂 

So I hope can add an option to ignore the statement like example above, thanks

### Suggested solution

I think can add `ignoreMultilineVariables` or `allowMultilineVariables` as a option for this purpose

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by niquenen** [comment]
Since the version [5.2.0](https://github.com/eslint-stylistic/eslint-stylistic/releases/tag/v5.2.0), TypeScript types are checked by `no-extra-parens`:

> - no-extra-parens:
>   - Support part of TS Nodes  -  by @9romise in https://github.com/eslint-stylistic/eslint-stylistic/pull/873 [(7d4aa)](https://github.com/eslint-stylistic/eslint-stylistic/commit/7d4aa27c)

Like @Yong-Zen-Xiang, we're using parentheses around types (unions, intersections, …) on multilines to improve the readability, for example:

```ts
type TFoo =
    First
    & Second
    & Third;

// We're using this
type TBar = (
    First
    & Second
    & Third
);
```

Furthermore, we're using parentheses to keep complex type more readable on declaration.

```ts
const foo: (
    MyMultilineVeryComplexType
) = {};
```

A setting to ignore multilines (or to ignore TypeScript types) would be a great idea.
